### ISSUE TITLE
Back button on some group page is not working after one or more times clicking add assignments buttons on the same page

### ISSUE THREAD

**Issue by priyanshagra** [CircuitVerse/CircuitVerse#4599]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Back button on some group page is not working after one or more times clicking add assignments buttons on the same page

https://github.com/CircuitVerse/CircuitVerse/assets/127022024/262725eb-a361-4bff-98c9-c868e359ecb0

### Steps To Reproduce

1. Go to main site
2. Go to my groups section
3. go to view section 
4.click on back button (it is not working after  one or more times clicking other buttons on the same page)

### Expected Behavior

It should redirect back to all group listing section

### Device Information [optional]

```markdown
- OS:
- Browser:
- version:
```

### Are you working on this issue?

Yes

**Comment by priyanshagra** [comment]
@tanmoysrt please assign this issue to me

**Comment by priyanshagra** [comment]
@VaibhavUpreti @tachyons @tanmoysrt please assign this me this issue

**Comment by tanmoysrt** [comment]
Hi @priyanshagra you can work on the issue.

**Comment by Ayush-Vish** [comment]
@priyanshagra  Are you still working on this issue  ?

**Comment by Pulkit0729** [comment]
I think the issue description is not accurate, the back button is working but it is redirecting to wrong page and the expected behaviour is that it redirects to all groups page. @priyanshagra

**Comment by Aaditya-2407** [comment]
Hi @tanmoysrt and @tachyons,

I’m interested in picking this issue up. I noticed that the existing PR (#4684) has been stale for a long time and is currently blocked by several CI test failures and security flags.

I can provide a clean, updated implementation using the javascript:history.back() approach while ensuring all internationalization (i18n) tags are correct and the CI tests pass on the current master branch. 

Is it okay if I open a new PR for this?
